Upgrade MSSQL Docker image

wjglerum opened the pull request:
Updates the Docker image for MSSQL to `2022-latest` and sets the default suggested password with the required complexity (one of the tests failed on this), see https://hub.docker.com/_/microsoft-mssql-server

> MSSQL_SA_PASSWORD is the database system administrator (userid = 'sa') password used to connect to SQL Server once the container is running. Important note: This password needs to include at least 8 characters of at least three of these four categories: uppercase letters, lowercase letters, numbers and non-alphanumeric symbols.

For M1 Macs we update the image to `azure-sql-edge:latest` as this now seems to work (verified locally), see https://hub.docker.com/_/microsoft-azure-sql-edge

wjglerum committed:
Upgrade MSSQL Docker image

Sanne commented:
Thanks @wjglerum ! LGTM

yrodiere reviewed (approved):
LGTM, though we'll probably want to mention in the migration guide the change of the default version used by dev services?

wjglerum commented:
> LGTM, though we'll probably want to mention in the migration guide the change of the default version used by dev services?

Good point! As people will probably have to update their own `license-acceptance.txt` files

gsmet reviewed (approved):
Thanks for the very thorough changes!

gsmet commented:
Let's see what CI has to say about this one.

gsmet commented:
The `ChangingCredentialsTest` failure is related.

wjglerum committed:
Don't use the username in the password

https://learn.microsoft.com/en-us/sql/relational-databases/security/strong-passwords?view=sql-server-ver16

wjglerum commented:
Turns out you can't use the username in the password anymore 😅 
https://learn.microsoft.com/en-us/sql/relational-databases/security/strong-passwords?view=sql-server-ver16
> Is not the name of a user.

gsmet reviewed (approved):
Thanks! I will add a note in the ~pull request~ migration guide.
